tests/integration/test_charm.py

```diff
@@ -175,3 +181,72 @@ async def test_prometheus_grafana_integration(ops_test: OpsTest):
     wait=tenacity.wait_exponential(multiplier=1, min=1, max=10),
     reraise=True,
 )
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_and_upgrade(ops_test: OpsTest):
```

Comment:
At first the purpose of this test case was not very clear to me, let's be more clear in the docstrings and change the name of the test case: `test_remove_redeploy_upgrade`, or similar.

Reply:
Yes. I am not comfortable with it as well. I will separate into two tests: remove and upgrade. We can have more testing on remove, if needed.